### DISCUSSION TITLE
Could Intel Mac package .app application for M1 Mac?

### DISCUSSION THREAD

**Discussion by shaucky** [airsdk/Adobe-Runtime-Support#2383] [Q&A]
Hello guys,

Last week i packaged a easy program and distribute to a gamer community. Then someone told me the Mac ver can't run on his M1 arch Macintosh. It alert "file is corrupt" (translated from chinese).

The SDK version i used is 50.1.1.2. I packaged the application with the tool include in Adobe Animate. My Mac is Intel i5 arch.

So yesterday i've read AIR SDK release notes about version 33.1.1.476, accroding to it i set the `adt.cfg` file with `DefaultArch=armv8` and `OverrideArch=armv8`, then packaged again. But he feedback me that the new packaged one also alert "file is corrupt".

Was there a problem with my way?

Thank you all.

----------------------------------------
update:

![IMG_9059](https://user-images.githubusercontent.com/119972730/209658811-4879fd49-378f-441e-a221-3c0bf3700253.JPG)

This is a screen the user gave me. 

The translation is:

"xxx.app" is corrupted and cannot be opened. You should push out the disk image.
This item is located on the disk image "xxx.dmg". This disk image was downloaded from Baidu WebDisk on December 25, 2022.
【Push out disk image】          【Cancel】

**Comment by JohnBlackburne** [comment]
Intel apps should and do run on M1 Macs. At some point possibly soon Apple will remove support for them, but right now they work and work pretty well, for emulated apps.

"file is corrupt" sounds more like the error you get with an unsigned app. Are you a registered Apple developer? If not your app will be rejected on other people's Macs, as it's not from a recognised developer. That's fine though, many apps including some from commercial developers have this issue.

The fix is something they need to do. Instead of using double-click to open your app, tell them to right-click and choose "Open" from the menu that appears. When they do this the dialog warning them about the app has an extra button to open it, which bypasses Apple's warnings. They only need to do this once, as the OS remembers and treats the app normally after that.

**Reply by shaucky** [reply]
The user told me it didn't work, i asked him for a screen and put it on the comment. If you have any idea please tell me.

Thanks a lot.

**Comment by JohnBlackburne** [comment]
That definitely looks like the dialog it puts up for an unsigned app, from an unrecognised developer.

 If you want to compare it yourself the Mac version of Realm of the Mad God has exactly this problem, in two places, the disk image and installer. You can download it from [here](https://www.realmofthemadgod.com).

Again this is not a problem; not too long ago all Mac apps were unsigned. Apple introduced Mac app signing like on iOS but it's **never been a requirement**. Users can always run unsigned apps. One workaround is the one I described above, right-click instead of double click. Another is once they see that dialog visit the Privacy & Security panel in System Settings where there should be an option to open the app.

**Comment by JohnBlackburne** [comment]
OK, what I wrote was wrong. I realised I could compare it myself, by switching my own M1 to Chinese and launching an app I know isn't properly signed. And although the dialog is superficially the same the contents are different.

<img width="372" alt="截屏2022-12-27 15 36 01" src="https://user-images.githubusercontent.com/6003194/209688945-e3ef6509-9035-48c3-b166-5ce7d168683a.png">

**Reply by shaucky** [reply]
As a Mac user, i also seen unsigned application like this, so when u commented the problem is caused by unsigned, i'm not sure yet.

So u neither sure how applications for Intel Mac will perform on M1 Mac, right? However, thanks for your kindness.

I've search for some information about this these days, seems .app application for Intel Mac can run on M1 Mac by some way. But how would Macintosh alert user, i'm not sure. But i guess it's different from "file is corrupt".

**Comment by JohnBlackburne** [comment]
I tried downloading your app myself from pan.bai*** but lack the Chinese phone number I need to register an account there. You might upload it somewhere it can be more easily accessed.
